presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/writers/ColumnWriters.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.writers;
+
+import com.facebook.presto.spi.type.Type;
+
+import static com.facebook.presto.spi.type.StandardTypes.BIGINT;
+import static com.facebook.presto.spi.type.StandardTypes.BOOLEAN;
+import static com.facebook.presto.spi.type.StandardTypes.CHAR;
+import static com.facebook.presto.spi.type.StandardTypes.DATE;
+import static com.facebook.presto.spi.type.StandardTypes.DOUBLE;
+import static com.facebook.presto.spi.type.StandardTypes.HYPER_LOG_LOG;
+import static com.facebook.presto.spi.type.StandardTypes.INTEGER;
+import static com.facebook.presto.spi.type.StandardTypes.P4_HYPER_LOG_LOG;
+import static com.facebook.presto.spi.type.StandardTypes.TIME;
+import static com.facebook.presto.spi.type.StandardTypes.TIMESTAMP;
+import static com.facebook.presto.spi.type.StandardTypes.TINYINT;
+import static com.facebook.presto.spi.type.StandardTypes.VARBINARY;
+import static com.facebook.presto.spi.type.StandardTypes.VARCHAR;
+
+public final class ColumnWriters
+{
+    public static final int DEFAULT_INITIAL_CAPACITY = 1024;
```

Comment:
This capacity is currently only used when converting keys for index join. It converts data from a cursor, so it doesn't know the number of records upfront. All remaining places specify the size directly, so no wasted space there. I've heard, 1024 is the number of index join keys in a batch, so this is how it was picked. In any case, it seems fine, because it shouldn't be too small to avoid re-allocations and shouldn't be too large to avoid wasted space.